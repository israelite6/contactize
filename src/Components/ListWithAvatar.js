import React, {useCallback, useState} from 'react';
import PropTypes from 'prop-types';
import {TouchableOpacity, StyleSheet, Text} from 'react-native';
import Avatar from './Avatar';

/**
 *
 * @param {string} label - Avatar directory
 */
export default function ListWithAvatar({source, label, onChange}) {
  const [isOpen, setIsOpen] = useState(false);

  const handleToggle = useCallback(
    (toggle) => () => {
      setIsOpen(toggle);
      onChange(toggle);
    },
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [],
  );

  return (
    <TouchableOpacity style={styles.root} onPress={handleToggle(!isOpen)}>
      <Avatar source={source} />

      <Text style={styles.label}>{label}</Text>
      <Text style={{transform: [{rotate: isOpen ? '0deg' : '90deg'}]}}>^</Text>
    </TouchableOpacity>
  );
}

ListWithAvatar.propTypes = {
  source: PropTypes.string,
  label: PropTypes.string.isRequired,
};

ListWithAvatar.defaultProps = {
  source: null,
  label: 'Name',
};

const styles = StyleSheet.create({
  root: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  label: {paddingLeft: 10, flexGrow: 1},
});
