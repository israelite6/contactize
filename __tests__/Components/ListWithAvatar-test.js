/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ListWithAvatar from '../../src/Components/ListWithAvatar';

it('ListWithAvatar component renders correctly', () => {
  const tree = renderer.create(<ListWithAvatar />).toJSON();
  expect(tree).toMatchSnapshot();
});
