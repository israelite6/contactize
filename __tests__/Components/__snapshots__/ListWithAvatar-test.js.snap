// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListWithAvatar component renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 30,
        "height": 60,
        "width": 60,
      }
    }
  />
  <Text
    style={
      Object {
        "flexGrow": 1,
        "paddingLeft": 10,
      }
    }
  >
    Name
  </Text>
  <Text
    style={
      Object {
        "transform": Array [
          Object {
            "rotate": "90deg",
          },
        ],
      }
    }
  >
    ^
  </Text>
</View>
`;
